pkg/serviceprovider/github/github.go

```diff
@@ -122,6 +131,105 @@ func (g *Github) GetServiceProviderUrlForRepo(repoUrl string) (string, error) {
 	return serviceprovider.GetHostWithScheme(repoUrl)
 }
 
+func (g *Github) CheckRepositoryAccess(ctx context.Context, cl client.Client, accessCheck *api.SPIAccessCheck) *api.SPIAccessCheckStatus {
+	repoUrl := accessCheck.Spec.RepoUrl
+	publicRepo := g.publicRepo(ctx, accessCheck)
+	status := &api.SPIAccessCheckStatus{
+		RepoURL:         repoUrl,
+		Accessible:      publicRepo,
+		Private:         !publicRepo,
+		Type:            api.SPIRepoTypeGit,
+		ServiceProvider: api.ServiceProviderTypeGitHub,
+	}
+
+	lg := log.FromContext(ctx)
+
+	tokens, lookupErr := g.lookup.Lookup(ctx, cl, accessCheck)
+	if lookupErr != nil {
+		lg.Error(lookupErr, "failed to lookup token for accesscheck", "accessCheck", accessCheck)
```

Comment:
Maybe it would be better to actually return the error here so that the reconciler can retry?

Reply:
yes, fixed